lms/tasks/annotations.py

```diff
@@ -1,17 +1,105 @@
 import logging
 
+from pydantic import BaseModel
+from sqlalchemy import select
+
+from lms.models import Assignment, LMSUser
 from lms.tasks.celery import app
 
+
+class _AssignmentMetadata(BaseModel):
+    resource_link_id: str
+
+
+class _LMSMetadata(BaseModel):
+    guid: str
+    assignment: _AssignmentMetadata
+
+
+class _AnnotationMetadata(BaseModel):
+    lms: _LMSMetadata
+
+
+class _AnnotationMention(BaseModel):
+    userid: str
+
+
+class _AnnotationPermissions(BaseModel):
+    read: list[str]
+    admin: list[str]
+    update: list[str]
+    delete: list[str]
+
+    @property
+    def is_private(self) -> bool:
+        # Private annotations have the same read permissions as the rest, the userid.
+        return self.read[0] == self.admin[0] == self.update[0] == self.delete[0]
+
+
+class _Annotation(BaseModel):
+    id: str
+    user: str
+    permissions: _AnnotationPermissions
+    mentions: list[_AnnotationMention]
+    metadata: _AnnotationMetadata
+
+
+class AnnotationEvent(BaseModel):
+    action: str
+    annotation: _Annotation
+
+
 LOG = logging.getLogger(__name__)
 
 
 @app.task
-def annotation_event(*, event) -> None:  # noqa: ARG001
+def annotation_event(*, event) -> None:
     """
     Process annotations events.
 
     These are published directly on LMS's queue by H
     """
-    # For now we are just consuming the messages to keep the queue clean while
-    # we deploy the publishing side on H.
-    return
+    annotation_event = AnnotationEvent(**event)
+    annotation = annotation_event.annotation
+
+    if annotation.permissions.is_private:
+        LOG.info("Skipping private annotation %s", annotation.id)
+        return
+
+    guid = annotation.metadata.lms.guid
+    resource_link_id = annotation.metadata.lms.assignment.resource_link_id
+
+    with app.request_context() as request, request.tm:
+        db = request.db
+
+        mentioning_user = db.execute(
+            select(LMSUser).where(LMSUser.h_userid == annotation.user)
+        ).scalar_one()
```

Comment:
Also not handling users / assignment not existing. There's no scenario that we anticipate for that so it it does happen, again we want to find out with a loud crash.